vcd/resource_vcd_org_vdc.go

```diff
@@ -314,10 +327,32 @@ func resourceVcdVdcRead(d *schema.ResourceData, meta interface{}) error {
 
 	d.Set("storage_profile", adminVdc.AdminVdc.VdcStorageProfiles)
 
+	vdc, err := adminOrg.GetVdcByName(d.Get("name").(string))
+	if err != nil || adminVdc == (govcd.AdminVdc{}) {
+		log.Printf("[DEBUG] Unable to find VDC")
+		return fmt.Errorf("unable to find VDC %#v", err)
+	}
+	metaData, err := vdc.GetMetadata()
+	if err != nil {
+		log.Printf("[DEBUG] Unable to get VDC metadata")
+		return fmt.Errorf("unable to get VDC metadata %#v", err)
+	}
+
+	d.Set("metadata", getMetaDataStruct(metaData.MetadataEntry))
+
 	log.Printf("[TRACE] vdc read completed: %#v", adminVdc.AdminVdc)
 	return nil
 }
 
+// Converts to terraform understandable structure
+func getMetaDataStruct(metaData []*types.MetadataEntry) map[string]interface{} {
```

Comment:
Nit: please use `Metadata` capitalization everywhere as we're using "metadata" as a single word.